lib/services/hpcpackage_remain.pm

```diff
@@ -89,10 +91,14 @@ sub full_pkgcompare_check {
     my $stage = $hash{stage};
 
     if ($stage eq 'before') {
+        # It need python2 module to compare packages
+        add_suseconnect_product("sle-module-python2", undef, undef, undef, 300, 1) if (check_var('DROPPED_MODULES', 'python2'));
         list_pkg("orignalq1w2.txt");
         install_pkg();
         list_pkg("installe3r4.txt");
         compare_pkg("/tmp/orignalq1w2.txt", "/tmp/installe3r4.txt");
+        # De-register python2 module again
+        deregister_dropped_modules if (check_var('DROPPED_MODULES', 'python2'));
```

Comment:
It's better to deregister python2 directly since this is a workaround that python2 is added by yourself so no doubt.